app/assets/javascripts/host_edit.js

```diff
@@ -77,6 +75,22 @@ function computeResourceSelected(item){
   }
 }
 
+function handle_nic_updates() {
+  var modal_window = $('#interfaceModal');
+
+  var handler = function(event = null) {
+    update_nics(function() {
+      interface_subnet_selected(primary_nic_form().find('select.interface_subnet'));
+    });
+  };
+
+  if (modal_window.is(":visible")) {
+    modal_window.modal('hide').on('hidden.bs.modal', handler);
```

Comment:
@ares: Thank you for your comment. I checked and didn't  see the handler running twice. The modal window div gets destroyed in update_nics anyway. What's your suggestion to change this?